okhttp/src/main/java/okhttp3/internal/ws/WebSocketWriter.java

```diff
@@ -53,12 +54,15 @@
   final Buffer buffer = new Buffer();
   final FrameSink frameSink = new FrameSink();
 
+  final boolean compressionEnabled;
```

Comment:
shouldn’t this be a parameter of `writeMessageFrame` ? I don’t think WebSocketWriter wants to be limited to always compressed or always uncompressed. If that were the case the protocol wouldn’t have facilities for changing that on a per-message basis

Reply:
You're right. 

Actually, rfc [suggests](https://tools.ietf.org/html/rfc7692#section-7.3) the trick where messages can stay uncompressed if their deflated versions are bigger in size **and no** "context takeover" is used. Implemented.